src/System.Private.CoreLib/shared/System/SpanHelpers.Byte.cs

```diff
@@ -199,10 +200,22 @@ public static unsafe int IndexOf(ref byte searchSpace, byte value, int length)
             IntPtr index = (IntPtr)0; // Use IntPtr for arithmetic to avoid unnecessary 64->32->64 truncations
             IntPtr nLength = (IntPtr)length;
 
-            if (Vector.IsHardwareAccelerated && length >= Vector<byte>.Count * 2)
+            if (Avx2.IsSupported || Sse2.IsSupported)
             {
-                int unaligned = (int)Unsafe.AsPointer(ref searchSpace) & (Vector<byte>.Count - 1);
-                nLength = (IntPtr)((Vector<byte>.Count - unaligned) & (Vector<byte>.Count - 1));
+                if (length >= Vector128<byte>.Count * 2)
```

Comment:
Why do `Avx2` and `Sse` kick in for 32-bytes but `Vector<T>` kick in for 32-bytes or 64-bytes?

Reply:
`Avx2` falls through to `Sse` so can operate on both 64 and 32 at the same time.

The `Vector<T>` branch only operates on one size (if its the longer Vector it can't then move a shorter Vector) so doesn't have that flexibility.

Reply:
A brief comment here indicating the above would be useful. I think it is probably something that people familiar with `Vector<T>`/HWIntrinsics might question.